### PR TITLE
Accordion bug in Formation: Bug/18238

### PR DESCRIPTION
**Pull request by accbjt**
Accordion not working correctly on mobile. Nested div's do not get recognized.

this works:
```
<div class="usa-accordion">
   <button class="usa-accordion-button">
   <div class="usa-accordion-content">
```

this doesn’t, but used to:
```
<div class="usa-accordion">
   <div>
        <button class="usa-accordion-button">
    </div>
   <div class="usa-accordion-content">
```

### CODE REVIEW

File: packages/documentation/src/pages/platform/unit-testing.mdx [hunk 1]
```diff
@@ -0,0 +1,108 @@
+## How do unit tests work?
```
Comment (rianfowler) [inline packages/documentation/src/pages/platform/unit-testing.mdx]: looks like some unit test content made it into this PR. 

you can clear out these changes with something like `git checkout origin master packages/documentation/src/pages/platform/unit-testing.mdx`

just make sure these changes are on your unit test branch.

File: packages/formation/js/accordion.js [hunk 2]
```diff
@@ -26,9 +26,13 @@ const loadAccordionHandler = () => {
 
   usaAccordionContentElements.forEach(el => {
     if (el.getAttribute('aria-hidden') === null) {
+      const buttonElement = el.parentElement.querySelector(
```
Comment (rianfowler) [inline packages/formation/js/accordion.js]: could use `querySelectorAll` for some of this logic to make it more declarative. The query is kind of long but this logic is also really complicated.

```
// creates exactly the array you're looking for without using spread operator 
const usaAccordion = document.querySelectorAll('.usa-accordion, .usa-accordion-bordered')

// inverts the logic a little- looks for the button first and then queries the siblings
// the matching is done with the query e.g. where button's aria-expanded is true and .usa-accordion-content is not false 
document
.querySelectorAll('button[aria-controls][aria-expanded='true'] ~ .usa-accordion-content:not[aria-hidden='false')
.forEach(el => el.setAttribute(
	        'aria-hidden',
	        false,
	      );

document
.querySelectorAll('button[aria-controls][aria-expanded='false'] ~ .usa-accordion-content:not[aria-hidden='true')
.forEach(el => el.setAttribute(
	        'aria-hidden',
	        true,
	      );
```

I think this is mostly a matter of taste- I think query selectors for this kind of thing are easier to understand than a few lines of imperative code. Maybe @jbalboni has an opinion. 

Not totally relevant but `querySelectorAll` is supposed to be highly optimized.

Comment (accbjt) [reply packages/formation/js/accordion.js]: I didn't know you can even do that. Thanks

**Comment by jaredcunha-usds** [comment]
Hmmm - As is, this is still not working on an older version of the mobile nav. Doesn't seem to impact anything else.

**Comment by accbjt** [comment]
@jaredcunha-usds Can you share with me the html for the mobile nav your testing it on.

File: packages/formation/js/accordion.js [hunk 3]
```diff
@@ -16,60 +16,63 @@ const isElementInViewport = (
...
 
 const loadAccordionHandler = () => {
-  const usaAccordion = [
-    ...document.getElementsByClassName('usa-accordion'),
-    ...document.getElementsByClassName('usa-accordion-bordered'),
-  ];
-  const usaAccordionContentElements = [
-    ...document.getElementsByClassName('usa-accordion-content'),
-  ];
+  document
```
Comment (accbjt) [inline packages/formation/js/accordion.js]: @rianfowler I used the query select but couldn't use your selector due to the situation when a button is nested in an element as how it is in mobile view on vets. The sibling combinator wouldn't work.  I couldn't find a selector that would work with nested children, so this is what I came up with. Any suggestions are welcomed.

File: packages/formation/js/accordion.js [hunk 4]
```diff
@@ -16,60 +16,63 @@ const isElementInViewport = (
...
+    });
 
-  usaAccordion.forEach(element => {
-    element.addEventListener('click', e => {
-      const target = e.target;
-      const other = Array.from(
-        element.getElementsByClassName('usa-accordion-button'),
-      ).filter(item => item !== target);
+  document
+    .querySelectorAll('.usa-accordion, .usa-accordion-bordered')
```
Comment (accbjt) [inline packages/formation/js/accordion.js]: This is a lot better then using the spread operator. I didn't know querySelectorAll returned an element that had a forEach method in it. Thank @rianfowler

**Comment by accbjt** [comment]
@jaredcunha-usds This should work now. I tested it on your repo.

**Comment by jaredcunha-usds** [comment]
Still not opening: 
```
TypeError: e.parentElement.querySelector(...) is null[Learn More] formation.js:1:1638
    o http://localhost:4000/vendor/javascripts/formation.js:1
    forEach self-hosted:262
    o http://localhost:4000/vendor/javascripts/formation.js:1
```

**Comment by accbjt** [comment]
@jaredcunha-usds yeah I must have added back the old code by accident when refactoring . Try it now.

**Comment by jaredcunha-usds** [comment]
Ah okay that seems to work now!

**Comment by jaredcunha-usds** [comment]
I think it' ll just need to be bumped to `6.2.4`

**Pull request merged**